modules/cert-manager-acme-http01.adoc

```diff
@@ -0,0 +1,121 @@
++
+// TODO: Still could use more details on what exactly this means
+
+.Procedure
+
+. Create an ACME cluster issuer.
+
+.. Create a YAML file that defines the `ClusterIssuer` object:
++
+.Example `acme-cluster-issuer.yaml` file
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-staging                                        <1>
+spec:
+  acme:
+    preferredChain: ""
+    privateKeySecretRef:
+      name: <secret_for_private_key>                               <2>
+    server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+    solvers:
+      - http01:
+          ingress:
+            class: openshift-default                               <4>
+----
+<1> Provide a name for the cluster issuer.
+<2> Replace `<secret_private_key>` with the name of secret to store the ACME account private key in.
+<3> Specify the URL to access the ACME server's `directory` endpoint. This example uses the _Let's Encrypt_ staging environment.
+<4> Specify the ingress class. This must be `openshift-default`.
+
+.. Create the `ClusterIssuer` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f acme-cluster-issuer.yaml
+----
+
+. Create an ingress to expose the service of the user workload.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-ingress-namespace <1>
+----
+<1> Specify the namespace for the ingress.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+.. Create a YAML file that defines the `Ingress` object:
++
+.Example `ingress.yaml` file
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: sample-ingress                                           <1>
+  namespace: my-ingress-namespace                                <2>
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt-staging          <3>
+    acme.cert-manager.io/http01-ingress-class: openshift-default <4>
+spec:
+  ingressClassName: openshift-default                            <5>
+  tls:
+   - hosts:
+       - <ingress_domain_name>                                   <6>
+     secretName: sample-tls                                      <7>
+  rules:
+  - host: <ingress_domain_name>                                  <8>
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: sample-workload                                <9>
+            port:
+              number: 80
+----
+<1> Specify the name of the ingress.
+<2> Specify the namespace that you created for the ingress.
+<3> Specify the cluster issuer that you created.
+<4> Specify the ingress class from the cluster issuer.
+<5> Specify the ingress class from the cluster issuer.
```

Comment:
`IngressClass` is a separate resource. The ingressclass `openshift-default` is created by the cluster-ingress-operator which is associated with the default ingresscontroller resource. This `ingressclass` is referenced in both the `ClusterIssuer` and the `Ingress`. 

If the user has a different ingresscontroller created say `secondary` which they want to use for HTTP-01, then the `IngressClass` they need to mention would be `openshift-secondary`  ([ref](https://github.com/openshift/cluster-ingress-operator/blob/a29464eda9614ac3d70c185b572f4839195e7306/pkg/operator/controller/names.go#L262)).

I think we should add this information just to make things clear both here and even on the `ClusterIssuer` sample.

Reply:
@thejasn I'm not quite following. I thought that, at least in the cluster issuer resource, that it was required for the ingress class to be "openshift-default". Is that not true?

If it can be different, can/would be different the whole way throughout the procedure, for both the cluster issuer and issuer resources? For example, if they wanted to use that "secondary" ingress controller instead of the default one, would they use "openshift-secondary" in all 3 places that it appears in the config now, or just in some of them?

(The 3 places being `spec.acme.solvers.http01.ingress.class` in the cluster issuer, and then the `acme.cert-manager.io/http01-ingress-class` annotation and `spec.ingressClassName` in the ingress?

Reply:
Yes, if the user wishes to use a different ingresscontroller (the `secondary` ingresscontroller) they would need to mention `openshift-secondary` in all 3 places.

Reply:
Replied in another thread. Kept this openshift-default and generic description